### PR TITLE
assertion/optimised mode

### PR DESCRIPTION
**Pull request by maxfischer2781**
Documents and specifies the use of ``assert`` and related functionality. This pull request includes:

* documentation for regular and optimised invocation (Python ``-O`` flag)
* a helper to cover assertion-only unittests,
* unittests to ensure the same behaviour in regular and optimised mode,
* travis runs unittests in both regular and optimised mode.

This pull request closes issue #8.

### CODE REVIEW

File: docs/source/topics/debug.rst [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+In order to help writing correct simulations,
+μSim uses a wide range of usage assertions.
+These generally verify argument values and types in advance,
+so that you can pinpoint where errors originate.
+
+At the same time, μSim strives to provide
+best performance for correct simulations.
+Once you have verified your simulation,
+you can elide all consistency assertions with zero cost.
```
Comment (eileen-kuehn) [inline docs/source/topics/debug.rst]:
```suggestion
you can omit all consistency assertions with zero cost.
```

File: docs/source/topics/debug.rst [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    >>> from usim import time, run
+    >>>
+    >>> async def wait(delay):
+    ...     await (time + delay)
+    ...
+    >>> run(wait(-20))  # delay until a time that has already passed
+
+When run regularly, this will fail with an :py:exc:`AssertionError`.
+The error includes some information to tell you what went wrong,
+and attempt to explain how to fix it.
```
Comment (eileen-kuehn) [inline docs/source/topics/debug.rst]:
```suggestion
and attempts to explain how to fix it.
```

File: docs/source/topics/debug.rst [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+an Exception with additional help on intended usage:
+
+.. code:: none
+
+    TypeError: 'time' cannot be used in 'await' expression
+
+    Use 'time' to derive operands for specific expressions:
+    * 'await (time + duration)' to delay for a specific duration
+    * 'await (time == date)' to delay until a specific point in time
+    * 'await (time >= date)' to delay until after a point in time
```
Comment (eileen-kuehn) [inline docs/source/topics/debug.rst]:
```suggestion
    * 'await (time >= date)' to delay until a point in time has passed
```

File: docs/source/topics/debug.rst [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+    ...     risk.recover()
+    ... except AssertionError:  # incorrect - recover from corruption
+    ...     risk.recover()
+
+When an assertion of μSim fails, this means your simulation
+is in an undefined state.
+Instead of trying to recover, you should fix the root cause
+of the erroneous condition.
+
+Eliding Assertions
```
Comment (eileen-kuehn) [inline docs/source/topics/debug.rst]:
```suggestion
Omitting Assertions
```

File: docs/source/topics/debug.rst [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+------------------
```
Comment (eileen-kuehn) [inline docs/source/topics/debug.rst]:
```suggestion
-------------------
```

File: docs/source/topics/debug.rst (hunk 3 above)
Comment (maxfischer2781) [reply docs/source/topics/debug.rst]: This is a copy of the original error message. Should we change that one as well?

Comment (eileen-kuehn) [reply docs/source/topics/debug.rst]: You can also just ignore. It just reads a bit strange... You can decide.

**Pull request merged**